pkg/operator/baremetal_config.go

```diff
@@ -9,7 +11,17 @@ import (
 )
 
 var provisioningGVR = schema.GroupVersionResource{Group: "metal3.io", Resource: "provisionings", Version: "v1alpha1"}
-var baremetalProvisioningCR = "cluster"
+var baremetalProvisioningCR = "provisioning-configuration"
+var unknownProvisioningConfig = "unknown"
```

Comment:
Nit: Not a fan of magic values like this, I think you'd be better off having your functions return a pointer, and returning `nil` if no value is found

Reply:
Yup, return value to a pointer is better here when an empty string is also a valid value.